### PR TITLE
DM-7303: Port to Python 3

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: tests/testStrayFlux.py [hunk 1]
```diff
@@ -31,6 +32,7 @@
 import lsst.pex.logging as pexLogging
 from lsst.meas.deblender.baseline import deblend
 import lsst.meas.algorithms as measAlg
+from functools import reduce
```
Comment (timj) [inline tests/testStrayFlux.py]: I think this needs to go above the LSST imports.

File: examples/designdoc.py [hunk 2]
```diff
@@ -32,7 +33,8 @@ def main():
     parser.add_option('--ccd', dest='ccd', type=int, default=5, help='Suprimecam CCD number')
     parser.add_option('--prefix', dest='prefix', default='design-', help='plot filename prefix')
     parser.add_option('--suffix', dest='suffix', default=None, help='plot filename suffix (default: ".png")')
-    parser.add_option('--pat', dest='pat', help='Plot filename pattern: eg, "design-%(pid)04i-%(name).png"; overrides --prefix and --suffix')
+    parser.add_option('--pat', dest='pat',
+        help='Plot filename pattern: eg, "design-%(pid)04i-%(name).png"; overrides --prefix and --suffix')
```
Comment (timj) [inline examples/designdoc.py]: indenting?

File: python/lsst/meas/deblender/baseline.py [hunk 3]
```diff
@@ -260,11 +268,11 @@ def deblend(footprint, maskedImage, psf, psffwhm,
     if not strayFluxToPointSources in validStrayPtSrc:
         raise ValueError((('strayFluxToPointSources: value \"%s\" not in the '
                            + 'set of allowed values: ')
```
Comment (timj) [inline python/lsst/meas/deblender/baseline.py]: I don't think this `+` is needed.

File: python/lsst/meas/deblender/baseline.py [hunk 4]
```diff
@@ -260,11 +268,11 @@ def deblend(footprint, maskedImage, psf, psffwhm,
...
-                           % strayFluxToPointSources) + str(validStrayPtSrc))
+                          % strayFluxToPointSources) + str(validStrayPtSrc))
     if not strayFluxAssignment in validStrayAssign:
         raise ValueError((('strayFluxAssignment: value \"%s\" not in the '
                            + 'set of allowed values: ')
```
Comment (timj) [inline python/lsst/meas/deblender/baseline.py]: `+` not needed.

File: tests/testStrayFlux.py (hunk 1 above)
Comment (fred3m) [inline tests/testStrayFlux.py]: ok

File: examples/designdoc.py (hunk 2 above)
Comment (fred3m) [inline examples/designdoc.py]: I think this was to limit the line length to under 110 chars. I could split the string into two lines but I think it looks better the way it is.

File: python/lsst/meas/deblender/baseline.py (hunk 3 above)
Comment (fred3m) [inline python/lsst/meas/deblender/baseline.py]: ok

File: python/lsst/meas/deblender/baseline.py (hunk 4 above)
Comment (fred3m) [inline python/lsst/meas/deblender/baseline.py]: ok

**Pull request merged**